Cargo.toml

```diff
@@ -9,21 +9,44 @@ dx12 = ["gfx-backend-dx12"]
 vulkan = ["gfx-backend-vulkan"]
 metal = ["gfx-backend-metal"]
 
-[[bin]]
+[lib]
 name = "ocean"
-path = "src/main.rs"
+path = "src/lib.rs"
+
+[[example]]
+name = "ocean_ios"
+path = "examples/ios/ios.rs"
+crate-type = ["staticlib"]
+required-features = ["metal"]
+
+[[example]]
+name = "ocean_vulkan"
+path = "examples/desktop.rs"
+required-features = ["vulkan"]
+
+[[example]]
+name = "ocean_metal"
+path = "examples/desktop.rs"
+required-features = ["metal"]
+
+[[example]]
+name = "ocean_dx12"
+path = "examples/desktop.rs"
+required-features = ["dx12"]
```

Comment:
is this useful to have separate examples for backends? You still can't make them *imply* a required feature, i.e.
`cargo run --example ocean_vulkan --features vulkan` is not better than just `cargo run --features vulkan`